projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -0,0 +1,202 @@
+		const wrongId = supportId.match( /([0-9]*)-(zd|chat)/i );
+		if ( wrongId ) {
+			const correctedId = `${ wrongId[ 1 ] }-${
+				wrongId[ 2 ].toLowerCase() === 'zd' ? 'zen' : 'hc'
+			}`;
+			correctedSupportIds.add( correctedId );
+		} else {
+			correctedSupportIds.add( supportId.toLowerCase() );
+		}
+	} );
+
+	return [ ...correctedSupportIds ];
+}
+
+/**
+ * Creates or updates a comment on issue.
+ *
+ * @param {WebhookPayloadIssue} payload - Issue event payload.
+ * @param {GitHub} octokit              - Initialized Octokit REST client.
+ * @param {Array} issueReferences       - Array of support references.
+ * @param {Array} issueComments         - Array of all comments on that issue.
+ */
+async function createOrUpdateComment( payload, octokit, issueReferences, issueComments ) {
+	const { issue, repository } = payload;
+	const { number } = issue;
+	const { name: repo, owner } = repository;
+	const ownerLogin = owner.login;
+
+	const existingComment = await getListComment( issueComments );
+
+	// If there is a comment already, update it.
+	if ( existingComment !== 0 ) {
+		debug(
+			`gather-support-references: update comment ID ${ existingComment } with our new list of references.`
+		);
+
+		const {
+			data: { body: existingCommentBody },
+		} = await octokit.rest.issues.getComment( {
+			owner: ownerLogin,
+			repo,
+			comment_id: existingComment,
+		} );
+
+		// First, build a list of all references and whether they are checked or not.
+		const listWithStatusMatch = existingCommentBody.matchAll(
+			/(?:-\s\[(?<checked>\s|x)\]\s)(?<ticketId>[0-9]*-(?:hc|zen))/gm
+		);
+		const statusMatches = [ ...listWithStatusMatch ];
+
+		// Build a new array with only checked ticket references.
+		const checkedList = [];
+		for ( const referenceStatus of statusMatches ) {
+			const [ , checked, ticketId ] = referenceStatus;
+			if ( checked === 'x' ) {
+				checkedList.push( ticketId );
+			}
+		}
+
+		// Remove checked tickets from our existing list of issue references.
+		const uncheckedList = issueReferences.filter(
+			reference => ! checkedList.includes( reference )
+		);
+
+		// Build our comment body, with first the checked references, then the unchecked references.
+		const updatedComment = `**Support References**
+${ checkedList
+	.map(
+		checkedTicket => `
+- [x] ${ checkedTicket }`
+	)
+	.join( '' ) }${ uncheckedList
+			.map(
+				uncheckedTicket => `
+- [ ] ${ uncheckedTicket }`
+			)
+			.join( '' ) }
+
+`;
+
+		await octokit.rest.issues.updateComment( {
+			owner: ownerLogin,
+			repo,
+			body: updatedComment,
+			comment_id: +existingComment,
+		} );
+	} else {
+		// If no comment was published before, publish one now.
+		debug( `gather-support-references: Posting comment to issue #${ number }` );
+
+		const comment = `**Support References**
+${ issueReferences
+	.map(
+		reference => `
+- [ ] ${ reference }`
+	)
+	.join( '' ) }
+
+`;
```

Comment:
It's be nice not to mostly-duplicate this bit from above. You could do a structure like
```
const checkedRefs = new Set();
if ( existingComment ) {
    populate checkedRefs
}

const comment = ...;
if ( existingComment ) {
    update
} else {
    create
]
```
Or you could just make a `buildComment( issueReferences, checkedRefs )` function to call.